.gitignore

```diff
@@ -1,6 +1,61 @@
+# Compiled Object files, Static and Dynamic libs (Shared Objects)
```

Comment:
It combines recommended .gitignore files for Go, C++, _and_ Java, instead of maintaining separate gitignore files in each component's directory. It's a combination of what was already in the repo, and recommendations from https://github.com/github/gitignore.